### PR TITLE
Extracted catalog watch auto-configuration.

### PR DESCRIPTION
**Pull request by TYsewyn**
Needed in order to have that bean running in the correct application context when SC Config discovery is enabled.
Also fixes a bug where Kubernetes discovery was enabled when discovery was disabled with `spring.cloud.kubernetes.enabled=false`.
Relates to spring-cloud/spring-cloud-config#1229

### CODE REVIEW

File: docs/src/main/asciidoc/discovery-client.adoc [hunk 1]
```diff
@@ -55,7 +55,3 @@ spring.cloud.kubernetes.discovery.enabled=false
 
 Some Spring Cloud components use the `DiscoveryClient` in order to obtain information about the local service instance. For
 this to work, you need to align the Kubernetes service name with the `spring.application.name` property.
-
-Spring Cloud Kubernetes can also watch the Kubernetes service catalog for changes and update the
```
Comment (spencergibb) [inline docs/src/main/asciidoc/discovery-client.adoc]: Why is this removed?

File: spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/KubernetesCatalogWatchAutoConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.scheduling.annotation.EnableScheduling;
+
+/**
+ * Auto configuration for catalog watcher.
+ *
+ * @author Tim Ysewyn
+ */
+@Configuration
+@EnableScheduling
```
Comment (spencergibb) [inline spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/KubernetesCatalogWatchAutoConfiguration.java]: We decided not to do this.

Comment (TYsewyn) [reply spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/KubernetesCatalogWatchAutoConfiguration.java]: Do you remember why? This is the only service registry implementation that needs that annotation explicitly.

Comment (spencergibb) [reply spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/KubernetesCatalogWatchAutoConfiguration.java]: We chose to document it instead.

Comment (ryanjbaxter) [reply spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/KubernetesCatalogWatchAutoConfiguration.java]: See https://github.com/spring-cloud/spring-cloud-kubernetes/issues/385

Comment (TYsewyn) [reply spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/KubernetesCatalogWatchAutoConfiguration.java]: Thanks for the input! So what if we do it like it's done in the [ConsulCatalogWatchAutoConfiguration](https://github.com/spring-cloud/spring-cloud-consul/blob/master/spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulCatalogWatchAutoConfiguration.java)?
Then we don't need to explicitly add `@EnableScheduling`, and it will only be created when the watcher is enabled with `spring.cloud.kubernetes.discovery.catalog-services-watch.enabled`.

Maybe not for this PR, but could be a nice improvement for the next minor release.

Comment (spencergibb) [reply spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/KubernetesCatalogWatchAutoConfiguration.java]: Yup, not for this PR.
